### PR TITLE
fix GSS on android

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
We send custom user agent to have info about device, but we need some identifier

Ref: https://github.com/nextcloud/android/pull/4218

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by MorrisJobke** [comment]
But this could also redirect users with browsers that have „(Android)“ in it and make a login there impossible ... thus we would need to refine the Android constant instead to be unique but still identifiable

**Comment by tobiasKaminsky** [comment]
Do you have a better identifier? We check that "(Android)" must be at the end of UserAgent. 
I checked firefox and chrome and both do not end with this.

**Comment by tobiasKaminsky** [comment]
The Android part is already merged (but can be changed).
As this is for a customer who cannot login with Android this is a bit more urgent. 
I would like to do a 3.7.1 with this fix next week and thus we should agree soon on this PR.

File: lib/Master.php [hunk 1]
```diff
@@ -200,7 +200,8 @@ protected function redirectUser($uid, $password, $location, array $options = [])
 			[
 				IRequest::USER_AGENT_CLIENT_IOS,
 				IRequest::USER_AGENT_CLIENT_ANDROID,
-				IRequest::USER_AGENT_CLIENT_DESKTOP
+				IRequest::USER_AGENT_CLIENT_DESKTOP,
+				'/^.*\(Android\).*$/'
```
Comment (MorrisJobke) [inline lib/Master.php]: But this is not "at the end, because there is still an `.*` before the `$`.

**Comment by tobiasKaminsky** [comment]
Sorry for that.
The push three days did not went through, but was somehow rejected and I assumed everything is ok.
I rebased and now it is really only checking for "(Android)" at end of string.

**Comment by MorrisJobke** [comment]
cc @rullzer

**Comment by tobiasKaminsky** [comment]
@rullzer @MorrisJobke @schiessle can we then merge it and deploy?
Then I can create a new android version.

**Comment by MorrisJobke** [comment]
> @rullzer @MorrisJobke @schiessle can we then merge it and deploy?

Deploy to where?

**Comment by rullzer** [comment]
Appstore. I'm on it.

**Pull request merged**